src/app/hooks/useViewTracker/index.jsx

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useState, useContext, useRef } from 'react';
+import { useInView } from 'react-intersection-observer';
+// Polyfill IntersectionObsesrver, e.g. for IE11
+import 'intersection-observer';
+
+import { ServiceContext } from '#contexts/ServiceContext';
+import { RequestContext } from '#contexts/RequestContext';
+import { buildATIClickParams } from '#containers/ATIAnalytics/params';
+import { getComponentInfo } from '#app/lib/analyticsUtils/index';
+import { sendEventBeacon } from '#containers/ATIAnalytics/beacon';
+
+const VIEWED_DURATION_MS = 1000;
+
+const useViewTracker = ({ pageData, componentName, actionLabel } = {}) => {
+  let eventTrackingProps;
+  const requestContext = useContext(RequestContext);
+  const serviceContext = useContext(ServiceContext);
+  const timer = useRef(null);
+  const [viewSent, setViewSent] = useState(false);
+  const [ref, inView] = useInView({
+    threshold: 0.5,
```

Comment:
Would we want the threshold to be 100% for the individual promos in the Onward Journeys block?

Reply:
Not for this PR, but I suspect we'll want to change the threshold based on the component.

Reply:
I'm not sure we would. I think the requirement is that all components should be 50% in-view before we track the event. I'll ask JB though